src/datachain/lib/dc/datachain.py

```diff
@@ -712,10 +712,41 @@ def _resolve_checkpoint(
             and not checkpoints_reset
             and metastore.find_checkpoint(job.parent_job_id, _hash)
         ):
-            # checkpoint found → reuse dataset
+            # checkpoint found → find which dataset version to reuse
+
+            # Find dataset version that was created by any ancestor job
+            dataset_version = metastore.get_dataset_version_for_job_ancestry(
```

Comment:
can we add inside those implementations some logs to actually trace it then (if job was found or not), if found - print its id, etc

Reply:
I means Studio primarily btw, not datachain (or datachain if we know that it will be visible on backend in Studio)